patches/minecraft/net/minecraft/world/gen/ChunkGeneratorEnd.java.patch

```diff
@@ -25,26 +25,35 @@
 +        this.field_73212_b = ctx.getScale();
 +        this.field_185973_o = ctx.getIsland();
 +        this.field_185972_n = (MapGenEndCity) net.minecraftforge.event.terraingen.TerrainGen.getModdedMapGen(this.field_185972_n, net.minecraftforge.event.terraingen.InitMapGenEvent.EventType.END_CITY);
++        this.structureManager = net.minecraftforge.event.terraingen.TerrainGen.getAddedStructures(ChunkGeneratorEnd.class);
      }
  
      public void func_180518_a(int p_180518_1_, int p_180518_2_, ChunkPrimer p_180518_3_)
-@@ -128,6 +142,7 @@
+@@ -128,6 +143,7 @@
  
      public void func_185962_a(ChunkPrimer p_185962_1_)
      {
 +        if (!net.minecraftforge.event.ForgeEventFactory.onReplaceBiomeBlocks(this, this.chunkX, this.chunkZ, p_185962_1_, this.field_73230_p)) return;
          for (int i = 0; i < 16; ++i)
          {
              for (int j = 0; j < 16; ++j)
-@@ -173,6 +188,7 @@
+@@ -173,6 +189,7 @@
  
      public Chunk func_185932_a(int p_185932_1_, int p_185932_2_)
      {
 +        this.chunkX = p_185932_1_; this.chunkZ = p_185932_2_;
          this.field_73220_k.setSeed((long)p_185932_1_ * 341873128712L + (long)p_185932_2_ * 132897987541L);
          ChunkPrimer chunkprimer = new ChunkPrimer();
          this.field_73231_z = this.field_73230_p.func_72959_q().func_76933_b(this.field_73231_z, p_185932_1_ * 16, p_185932_2_ * 16, 16, 16);
-@@ -254,6 +270,10 @@
+@@ -182,6 +199,7 @@
+         if (this.field_73229_q)
+         {
+             this.field_185972_n.func_186125_a(this.field_73230_p, p_185932_1_, p_185932_2_, chunkprimer);
++            this.structureManager.generate(this.field_73230_p, p_185932_1_, p_185932_2_, chunkprimer);
```

Comment:
Having `generate` and `generateStructure` could be a bit confusing -- maybe go with `primeStructure` and `generateStructure` instead?

Reply:
Just sticking to the Vanilla/MCP names which are a little bit confusing here. The structure manager  `generate` calls the added structures `generate`, same for the `generateStructure`.
So I think it makes sense this way.

I agree that generate could be renamed to avoid confusion, but I would suggest renaming the MapGenStructure method first